File: src/main/java/kr/co/lunatalk/global/common/response/GlobalResponse.java
package kr.co.lunatalk.global.common.response;


import java.time.LocalDateTime;

public record GlobalResponse(boolean isSuccess, int status, Object data, LocalDateTime timestamp) {
	public static GlobalResponse success(int status, Object data) {
		return new GlobalResponse(true, status, data, LocalDateTime.now());
	}

	public static GlobalResponse fail(int status, ErrorResponse errorResponse) {
		return new GlobalResponse(false, status, errorResponse, LocalDateTime.now());
	}
}

File: lunatalk-common/src/main/kotlin/kr/co/lunatalk/global/common/response/GlobalResponse.kt
package kr.co.lunatalk.global.common.response

import java.time.LocalDateTime

data class GlobalResponse(
    val isSuccess: Boolean,
    val status: Int,
    val data: Any?,
    val timestamp: LocalDateTime
) {
    companion object {
        fun success(status: Int, data: Any?): GlobalResponse =
            GlobalResponse(true, status, data, LocalDateTime.now())

        fun fail(status: Int, errorResponse: ErrorResponse): GlobalResponse =
            GlobalResponse(false, status, errorResponse, LocalDateTime.now())
    }
}
